jerry-core/vm/vm.c

```diff
@@ -2082,6 +2082,13 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         }
         case VM_OC_RETURN_PROMISE:
         {
+          left_value = ECMA_VALUE_UNDEFINED;
```

Comment:
How about adding `VM_OC_GET_STACK` flag for `CBC_EXT_RETURN_PROMISE`? `left_value` is always initialized to `ECMA_VALUE_UNDEFINED` in each loop which is suitable for `CBC_EXT_RETURN_PROMISE_UNDEFINED` .

Reply:
This is actually a rather clever idea!